lib/hanami/commands/generate/model.rb

```diff
@@ -103,11 +103,24 @@ def repository_path
         # @since 0.9.1
         # @api private
         def migration_path
+          existing_migration_path || new_migration_path
+        end
+
+        # @since 1.1.0
+        # @api private
+        def new_migration_path
           timestamp = Time.now.utc.strftime(Migration::TIMESTAMP_FORMAT)
-          filename = Migration::FILENAME_PATTERN % { timestamp: timestamp, name: "create_#{table_name}"}
+          filename = Migration::FILENAME_PATTERN % { timestamp: timestamp, name: migration_name_underscored }
+
           Hanami::Model.configuration.migrations.join(filename)
         end
 
+        # @since 1.1.0
+        # @api private
+        def existing_migration_path
+          Utils::FileList["#{Hanami::Model.configuration.migrations}/[0-9]*_#{migration_name_underscored}.rb"].first
```

Comment:
Since you're using `first` on the result you'd get the oldest migration which probably is always the one that was created by the generate model statement. Therefore this probably is not a real-world concern.